### PR TITLE
Add docs with CI

### PR DESCRIPTION
**Pull request by wsipak**
@tgorochowik 
The `Deploy` step in CI is executed only on master, so It hasn't been run yet. The docs are in `docs` directory.

### CODE REVIEW

File: .github/workflows/ghpages.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: GH Pages
+
+on: push
```
Comment (tgorochowik) [inline .github/workflows/ghpages.yml]: Please add a filter to only run on `master` and add the regular `pull_request` trigger, otherwise it won't run if the PR is made from a fork.

File: .github/workflows/ghpages.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          sudo apt-get install python3-sphinx
+          python3 -m pip install setuptools
+          python3 -m pip install git+https://github.com/antmicro/sphinx_antmicro_theme
+
+    - name: Build
+      run: |
+        make -C docs html
+
+    - name: Deploy
+      if: github.ref == 'refs/heads/master'
```
Comment (tgorochowik) [inline .github/workflows/ghpages.yml]: And then this should be sth like:
` if: github.ref == 'refs/heads/master' && github.event_name != 'pull_request'`
to make sure it runs only on push to master

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -27,7 +30,7 @@ jobs:
 
     - name: Lint with flake8
       run: |
-        flake8 . --count --show-source --statistics
+        flake8 fpga_topwrap --count --show-source --statistics
```
Comment (tgorochowik) [inline .github/workflows/ci.yml:33]: I think this should be a separate commit (and the change of the trigger above too)

File: .github/workflows/ghpages.yml (hunk 1 above)
Comment (wsipak) [reply .github/workflows/ghpages.yml]: Done

File: .github/workflows/ghpages.yml (hunk 2 above)
Comment (wsipak) [reply .github/workflows/ghpages.yml]: Done

File: .github/workflows/ci.yml (hunk 3 above)
Comment (wsipak) [reply .github/workflows/ci.yml:33]: This change is a separate commit now. The whole CI for GH Pages is a new file in a separate commit, too.

**Review by tgorochowik — APPROVED** [review]
